### ISSUE TITLE
[no-unused-vars] Importing type with same name as constructor reports unused definition

### ISSUE THREAD

**Issue by sweepline** [typescript-eslint/typescript-eslint#1971]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": "error",
  }
}
```

```TS
import { Service } from "...";

export class Whoop {
	constructor(private Service: Service) {
		this.Service.doSomething();
	}
}
```

**Expected Result**
No errors reported as we use the import for typing.

**Actual Result**
```
  1:10  error  'Service' is defined but never used  @typescript-eslint/no-unused-vars
```

**Additional Info**
This was [reported on my own plugin](https://github.com/sweepline/eslint-plugin-unused-imports/issues/4) which utilises this rules to autofix the unused imports, i tracked it to this issue reporting the import as unused.

**Versions**
I also tested `@typescript-eslint/eslint-plugin@canary`, same problem.

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.30.0` |
| `@typescript-eslint/parser`        | `2.30.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `6.13.1` |
| `npm`                              | `13.2.0` |

**Comment by bradzacher** [comment]
Please use the search

Duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/1856
